kafka-ui-react-app/src/components/Topics/List/List.tsx

```diff
@@ -41,25 +43,39 @@ const List: React.FC<Props> = ({
   const { isReadOnly } = React.useContext(ClusterContext);
   const { clusterName } = useParams<{ clusterName: ClusterName }>();
   const { page, perPage } = usePagination();
+  const [orderBy, setOrderBy] = React.useState<
+    TopicColumnsToSort | undefined
+  >();
+  const [search, setSearch] = React.useState<string>('');
+  let debounceTimeout: NodeJS.Timeout;
 
   React.useEffect(() => {
-    fetchTopicsList({ clusterName, page, perPage });
-  }, [fetchTopicsList, clusterName, page, perPage]);
+    fetchTopicsList({ clusterName, page, perPage, orderBy, search });
+  }, [fetchTopicsList, clusterName, page, perPage, orderBy, search]);
 
   const [showInternal, setShowInternal] = React.useState<boolean>(true);
 
   const handleSwitch = React.useCallback(() => {
     setShowInternal(!showInternal);
   }, [showInternal]);
 
+  const handleSearch = (e: React.ChangeEvent<HTMLInputElement>) => {
+    if (debounceTimeout) window.clearTimeout(debounceTimeout);
+    debounceTimeout = setTimeout(() => setSearch(e.target.value), 300);
+  };
+
+  React.useEffect(() => () => {
+    if (debounceTimeout) window.clearTimeout(debounceTimeout);
+  });
```

Comment:
You can use `useDebouncedCallback` (there's an example in `Messages` component).